### PR TITLE
Refactoring Block#getObject(...) to only return Block.class

### PR DESCRIPTION
**Pull request by dmitryvinn**
== NO RELEASE NOTE ==

### CODE REVIEW

**Comment by shixuan-fan** [comment]
Added reviewer based on github suggestions. Feel free to redirect.

**Comment by wenleix** [comment]
Thanks @dmitryvinn  for the contribution. The failure is due to Presto code-gen some code, and it's still calling `getObject` in these place.

For example
```
Caused by: java.lang.NoSuchMethodError: com.facebook.presto.spi.block.Block.getObject(ILjava/lang/Class;)Ljava/lang/Object;
	at com.facebook.presto.$gen.VarianceStateSerializer_20190922_044352_16045.deserialize(Unknown Source)
	at com.facebook.presto.$gen.DoubleBigintVarPopAccumulator_20190922_044352_16050.addIntermediate(Unknown Source)
	at com.facebook.presto.operator.Aggregator.processPage(Aggregator.java:61)
	at com.facebook.presto.operator.AggregationOperator.addInput(AggregationOperator.java:152)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:384)
	at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:283)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:675)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:276)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:483)
	at com.facebook.presto.$gen.Presto_0_227_SNAPSHOT_ee4b2e4__testversion____20190922_041207_6146.run(Unknown Source)
```

I think it's generated here:

https://github.com/prestodb/presto/blob/6a0db150f271631fa54bf01af622bd97dc234270/presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/StateCompiler.java#L247

Maybe change this line to be

```java
            deserializerBody.append(row.set(block.invoke("getObject", Block.class, index)));
```

?

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/StateCompiler.java [hunk 1]
```diff
@@ -244,7 +243,7 @@ else if (fields.size() == 1) {
         }
         else if (fields.size() > 1) {
             Variable row = scope.declareVariable(Block.class, "row");
-            deserializerBody.append(row.set(block.invoke("getObject", Object.class, index).cast(Block.class)));
+            deserializerBody.append(row.set(block.invoke("getObject", Object.class, index)));
```
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/StateCompiler.java]: What about? 

```
deserializerBody.append(row.set(block.invoke("getBlock", Block.class, index)))
```

Sorry I forgot to change `getObject` to `getBlock`, thus the method name still not found.

**Comment by dmitryvinn** [comment]
You are right, I should have renamed the entire method - I was worried about serialization even though it is not a serializable class!

What do you think about naming the naming #getInstance(index)? I think Block#getBlock might be a bit redundant. That being said, I don’t want users to think it is a singleton that tends to use #getInstance name

**Comment by wenleix** [comment]
@dmitryvinn : We can just call `getBlock`. It's getting a nested block inside a block (think about getting an element from array. :).

Also please squash all commits.

The tests turn green :D

**Comment by dmitryvinn** [comment]
Cool, I will rename the method and of course will squash the commits too!

**Pull request closed without merging**